.github/workflows/release.yaml

```diff
@@ -39,6 +32,43 @@ jobs:
             target/${{matrix.target}}/release/mirrord
             target/${{matrix.target}}/release/libmirrord_layer.${{matrix.extension}}
           if-no-files-found: error
+  build_binaries_macos:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          target: aarch64-apple-darwin
+      - uses: Swatinem/rust-cache@v1
+      - name: build mirrord-layer x86-64
+        run: cargo +nightly build --release -p mirrord-layer --target=x86_64-apple-darwin
+      - name: build mirrord-layer macOS arm
+        run: cargo +nightly build --release -p mirrord-layer --target=aarch64-apple-darwin
+      - name: create universal target dir
+        run: mkdir -p /tmp/target/universal-apple-darwin/release
+      - name: create universal dylib
+        run: lipo -create -output /tmp/target/universal-apple-darwin/release/libmirrord_layer.dylib target/aarch64-apple-darwin/release/libmirrord_layer.dylib target/x86_64-apple-darwin/release/libmirrord_layer.dylib
+      - name: build macOS arm cli with universal dylib
+        env:
+          MIRRORD_LAYER_FILE: /tmp/target/universal-apple-darwin/release/libmirrord_layer.dylib
+        run: cargo +nightly build --release -p mirrord --target=aarch64-apple-darwin
+      - name: build macOS x86-64 cli with universal dylib
+        env:
+          MIRRORD_LAYER_FILE: /tmp/target/universal-apple-darwin/release/libmirrord_layer.dylib
+        run: cargo +nightly build --release -p mirrord --target=x86_64-apple-darwin
+      - name: create universal macOS cli
+        run: lipo -create -output /tmp/target/universal-apple-darwin/release/mirrord target/aarch64-apple-darwin/release/mirrord target/x86_64-apple-darwin/release/mirrord
+      - name: change dir to tmp
+        run: cd /tmp
```

Comment:
I was under the impression that CWD is not preserved between steps?

Reply:
Good point, too bad upload artifacts is a very awful action.. I changed to copy the tmp target to current dir so it'll work now :)